frame/babe/src/lib.rs

```diff
@@ -781,12 +781,26 @@ impl<T: Config> frame_support::traits::EstimateNextSessionRotation<T::BlockNumbe
 		T::EpochDuration::get().saturated_into()
 	}
 
+	fn estimate_current_session_progress(_now: T::BlockNumber) -> Option<Percent> {
```

Comment:
I suppose for most use cases percent is fine, but FYI it should be easy enough to make this `estimate_current_session_progress<P: PerThing>(_: T::BlockNumer) -> Option<P>`, if you think it is desirable.

Reply:
(thinking about it myself again, probably not worth potential added complexity :D)

Reply:
Yeah I think it's not worth it, for our immediate use-case (im-online) even percentage has much more granularity than necessary since we're only interested in figuring out if we're past through half the session.